packages/convertkit/src/forms/subscribe/index.tsx

```diff
@@ -0,0 +1,136 @@
+}
+
+type SubscribeFormProps =
+  | ({
+      /**
+       * The tag id to tag a subscriber with.
+       * @type number
+       */
+      tag: number
+      form?: never
+      sequence?: never
+    } & SubscribeFormOptions)
+  | ({
+      tag?: never
+      /**
+       * The form id to subscribe to. Default is `process.env.NEXT_PUBLIC_CONVERTKIT_SIGNUP_FORM`.
+       * @type number
+       */
+      form: number
+      sequence?: never
+    } & SubscribeFormOptions)
+  | ({
+      tag?: never
+      form?: never
+      /**
+       * Sequence ID to subscribe to.
+       * @type number
+       */
+      sequence: number
+    } & SubscribeFormOptions)
+  | ({tag?: never; form?: never; sequence?: never} & SubscribeFormOptions)
+
+const SubscribeForm: React.FC<SubscribeFormProps> = ({
+  tag,
+  form,
+  sequence,
+  actionLabel = 'Subscribe',
+  onSuccessRedirectUrl = '/confirm',
+}) => {
+  const [isSubmitting, setSubmitting] = React.useState<boolean>(false)
+  const router = useRouter()
+
+  const formik = useFormik({
+    initialStatus: '',
+    initialValues: {
+      email_address: '',
+      first_name: '',
+    },
+    validationSchema: Yup.object().shape({
+      email_address: Yup.string()
+        .email('Invalid email address')
+        .required('Required'),
+      first_name: Yup.string(),
+    }),
+    enableReinitialize: true,
+    onSubmit: async (values) => {
+      setSubmitting(true)
+      axios
+        .post('/api/convertkit/subscribe', {...values, tag, form, sequence})
+        .catch((err) => {
+          formik.setStatus('error')
+          console.log(err)
+        })
+        .then((res: any) => {
+          if (onSuccessRedirectUrl) {
+            const url = queryString.stringifyUrl({
+              url: onSuccessRedirectUrl,
+              query: {
+                [CK_SUBSCRIBER_KEY]: res.data.id,
+                email_address: values.email_address,
+              },
+            })
+            router.push(url)
+          } else {
+            setSubmitting(false)
+            formik.setStatus('submitted')
+          }
+        })
+    },
+    validateOnChange: false,
+  })
+
+  return (
+    <form
+      data-sr-convertkit-subscribe-form={formik.status}
+      onSubmit={formik.handleSubmit}
+    >
+      {!formik.status && (
+        <>
+          <Input
+            label="First Name"
+            name="first_name"
+            onChange={formik.handleChange}
+            placeholder="Preferred name"
+            type="text"
+          />
+          <Input
+            label="Email"
+            name="email_address"
+            onChange={formik.handleChange}
+            placeholder="you@company.com"
+            type="email"
+            required
+          />
+          <Button isLoading={isSubmitting}>{actionLabel}</Button>
+        </>
+      )}
+      {formik.status === 'submitted' && <p>Thanks!</p>}
+      {formik.status === 'error' && <p>Something went wrong.</p>}
+    </form>
```

Comment:
and this is the UI part consisting of shared `Input` and `Button` atom components.